src/playlist.rs

```diff
@@ -373,8 +374,14 @@ impl SessionData {
     pub fn write_to<T: Write>(&self, w: &mut T) -> std::io::Result<()> {
         write!(w, "#EXT-X-SESSION-DATA:")?;
         write!(w, "DATA-ID=\"{}\"", self.data_id)?;
-        write!(w, ",VALUE=\"{}\"", self.value)?;
-        write!(w, ",URI=\"{}\"", self.uri)?;
+        // SessionData must contain either a VALUE or a URI,
+        // but not both https://tools.ietf.org/html/rfc8216#section-4.3.4.4
+        // TODO: This could be an enum of Value/Uri?
```

Comment:
Is this something you want to do in this PR? Its nice to have the logic as early as possible, so in the `from_hashmap` above. `from_hashmap` could then return a `Result<SessionData, String>`, that way it would even fail on parsing.

Reply:
I wanted to see what you think. I just did the quick hack here, because I'm not sure if introducing the extra enum subtype is worth it, or if there's a better way (I'm still pretty new at Rust)

Reply:
Yea i'd say an enum fits perfectly here if we go by the RFC, since it states it MUST be either VALUE or URI. My comment about having the parser abort is just expanding on the idea of having the enum type, but I think it's fine either way.

Reply:
I had a go at making `from_hashmap()` return a Result, but I'm not sure how to handle that in the caller code.

```
named!(pub session_data_tag<SessionData>,
    do_parse!( tag!("#EXT-X-SESSION-DATA:") >> attributes: key_value_pairs >> ( SessionData::from_hashmap(attributes)))                                                                                                                                                                                                               
);
```

Reply:
I see, yes macros make writing these things pretty tricky sometimes, which is why nom switched to functions in 5.0. In this case `map_res!` can be used to handle the result, something like:
```
named!(pub session_data_tag<SessionData>,
    do_parse!( tag!("#EXT-X-SESSION-DATA:") 
               >> session_data: map_res!(key_value_pairs, |attrs| SessionData::from_hashmap(attrs))
               >> session_data)
);
```

Reply:
Thanks :) I had to wrap the final `session_data` in an extra set of parentheses for unclear reasons, but that worked otherwise.